app/models/miq_group.rb

```diff
@@ -186,9 +203,29 @@ def miq_user_role_name
     miq_user_role.nil? ? nil : miq_user_role.name
   end
 
+  def system_group?
+    group_type == SYSTEM_GROUP
+  end
+
+  # @return true if this is a default tenant group
+  def tenant_group?
+    group_type == TENANT_GROUP
+  end
+
+  # Asks about the tenant's default_miq_group
+  #
+  # NOTE: this is the old definition for `tenant_group?`
+  #
+  # @return true if this is assigned to the tenant as the default tenant
+  # @return false if the tenant is being deleted or pointing to a different group
+  def referenced_by_tenant?
+    tenant.try(:default_miq_group_id) == id
+  end
+
   def read_only
-    group_type == "system"
+    system_group? || tenant_group?
   end
+  alias_method :read_only?, :read_only
```

Comment:
It was tricky for me to track down all the uses of `miq_group.read_only` in the code.
So this was me attempting to move towards a standard method.

I'll grep again, but not sure if I want to introduce this risk at this time.